### PR TITLE
doc: added release notes for v1.8 draft

### PR DESCRIPTION
**Pull request by Shilpashree3lc-zz**
Draft of Zephyr 1.8 release notes which contains some interesting
changes on Kernel, Architecture, Boards, Drivers and Sensors,
Networking, Bluetooth, documentation, and jira items.etc.

Signed-off-by: LC Shiplashree <shilpashree.lc@intel.com>

### CODE REVIEW

File: doc/release-notes-1.8.rst [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+* :jira:`ZEP-2211` - Makefile.toolchain.gccarmemb sets DTC to /usr/bin/dtc which is not available on macOS
+* :jira:`ZEP-2219` - Infinite loop caused by NA with unkown ICMPv6 option type (0xff)
+* :jira:`ZEP-2221` - irq_offload() unimplemented for Cortex M0 boards
+* :jira:`ZEP-2222` - [nrf 51] Zephyr fails to compile if TICKLESS_KERNEL is enabled for nrf51
+* :jira:`ZEP-2223` - samples: net: http_client: Cannot send OPTIONS/HEAD request on FRDM K64F
+* :jira:`ZEP-2224` - kernel/mem_pool: test_mpool_alloc_free_isr hits memory crash
+* :jira:`ZEP-2235` - Coverity static scan issues seen
+* :jira:`ZEP-2240` - kernel/test_build/test_newlib / fails for stm32373c_eval with ARM cross-compiler
+* :jira:`ZEP-2257` - test_context fails on bbc_microbit
+ :jira:`ZEP-472` -	Ethernet packets are getting missed if sent in quick succession.
```
Comment (dbkinder) [inline doc/release-notes-1.8.rst]: Not a valid list character -- use an asterisk (*) here and below

Also we'll need to generate this list again right before we do the release to catch any last-minute Jira issues (and to get a sorted list) -- the ones you've added aren't in sorted order with the others.

**Review by dbkinder — CHANGES_REQUESTED** [review]
list of Jira items needs updating (and correct formatting)

File: doc/release-notes-1.8.rst [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+* Xtensa GCC support
+* Initial implementation of MMU/MPU support
+* Expanded device support
+
+The following sections provide detailed lists of changes by component.
+
+Kernel
+******
+
+* Use k_cycle_get_32 instead instead of sys_cycle_get_32 for Kernel
```
Comment (cvinayak) [inline doc/release-notes-1.8.rst]: instead repeated twice

File: doc/release-notes-1.8.rst [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+* Added k_panic() and k_oops() APIs for Kernel
+* Added k_thread_create() API for Kernel
+* Added k_queue API for Kernel
+* Add tickless kernel support
```
Comment (cvinayak) [inline doc/release-notes-1.8.rst]: Added?

**Review by cvinayak — CHANGES_REQUESTED** [review]

**Review by nashif — CHANGES_REQUESTED** [review]
This needs to go to the 1.8 branch, not master

File: doc/release-notes-1.8.rst (hunk 3 above)
Comment (Shilpashree3lc) [inline doc/release-notes-1.8.rst]: Please elaborate your question.

**Pull request closed without merging**